Intersect.Client/Interface/Game/MapItem/MapItemWindow.cs

```diff
@@ -0,0 +1,167 @@
+﻿using System.Collections.Generic;
+
+using Intersect.Client.Core;
+using Intersect.Client.Framework.File_Management;
+using Intersect.Client.Framework.GenericClasses;
+using Intersect.Client.Framework.Gwen.Control;
+using Intersect.Client.Framework.Gwen.Control.EventArguments;
+using Intersect.Client.General;
+using Intersect.Client.Localization;
+using Intersect.GameObjects;
+
+namespace Intersect.Client.Interface.Game.Inventory
+{
+
+    public class MapItemWindow
+    {
+        //Item List
+        public List<MapItemIcon> Items = new List<MapItemIcon>();
+        private List<Label> mValues = new List<Label>();
+
+        //Controls
+        private ImagePanel mMapItemWindow;
+
+        private Label mMenuHeader;
+
+        private ScrollControl mItemContainer;
+
+        private Button mBtnLootAll;
+
+        //Init
+        public MapItemWindow(Canvas gameCanvas)
+        {
+            mMapItemWindow = new ImagePanel(gameCanvas, "MapItemWindow");
+            mMenuHeader = new Label(mMapItemWindow, "Title");
+            mMenuHeader.SetText(Strings.MapItemWindow.Title);
+
+            mItemContainer = new ScrollControl(mMapItemWindow, "ItemsContainer");
+            mItemContainer.EnableScroll(false, true);
+
+            mBtnLootAll = new Button(mMapItemWindow, "LootAllButton");
+            mBtnLootAll.Text = Strings.MapItemWindow.LootButton;
+            mBtnLootAll.Clicked += MBtnLootAll_Clicked;
+
+            mMapItemWindow.LoadJsonUi(GameContentManager.UI.InGame, Graphics.Renderer.GetResolutionString());
+
+            CreateItemContainer();
+        }
+
+        //Location
+        public int X => mMapItemWindow.X;
+
+        public int Y => mMapItemWindow.Y;
+
+        //Methods
+        public void Update()
+        {
+            // Is this disabled from the server config? If so, skip doing anything.
+            if (!Options.Loot.EnableLootWindow)
+            {
+                mMapItemWindow.Hide();
+                return;
+            }
+
+            var location = new Point(Globals.Me.X, Globals.Me.Y);
+            var mapItems = Globals.Me.MapInstance.MapItems;
+            if ((!mapItems.ContainsKey(location) || mapItems[location].Count < 1) && !mMapItemWindow.IsHidden)
+            {
+                mMapItemWindow.Hide();
+            }
```

Comment:
Needs to account for Globals.Me.MapInstance possibly being null (during warps between areas) and hiding the window in that case.

Reply:
```
            var mapItems = Globals.Me.MapInstance?.MapItems;
            if (mapItems == null || (!mapItems.ContainsKey(location) || mapItems[location].Count < 1) && !mMapItemWindow.IsHidden)
            {
                mMapItemWindow.Hide();
            }
            else if ( mapItems != null && mapItems.ContainsKey(location) && mapItems[location].Count > 0 && mMapItemWindow.IsHidden)
            {
                mMapItemWindow.Show();
            }



```